dev-packages/application-package/src/generator/backend-generator.ts

```diff
@@ -59,9 +59,11 @@ module.exports = (port, host) => Promise.resolve()${this.compileBackendModuleImp
     }
 
     protected compileMain(backendModules: Map<string, string>): string {
-        return ` // @ts-check
+        return `// @ts-check
 const serverPath = require('path').resolve(__dirname, 'server');
-module.exports = require('@theia/core/lib/node/cluster/main').default(serverPath);
+module.exports = require('@theia/core/lib/node/cluster/main').default(serverPath).then(function (address) {
```

Comment:
before it exported an instance of address, it looks that now it exports nothing, I think that electron relies on it in the dev mode.

Also this code is used for the browser and there `then` cause does not make sense.

Reply:
I will verify it once more to make sure it does not break existing behavior.

Reply:
I need do another iteration on this. You were right, it did not work in all cases. I have updated the PR description with the TODOs.